lxd/instance/drivers/driver_lxc.go

```diff
@@ -2522,6 +2531,94 @@ func (d *lxc) onStart(_ map[string]string) error {
 	return nil
 }
 
+// mountBpfFs mounts bpffs inside the container.
+func (d *lxc) mountBpfFs(pid int, bpffsParams map[string]string) error {
+	if !d.state.OS.BPFToken {
+		return fmt.Errorf("BPF Token mechanism is not supported by kernel running.")
+	}
+
+	pidFdNr, pidFd := seccomp.MakePidFd(pid, d.state)
+	if pidFdNr >= 0 {
+		defer func() { _ = pidFd.Close() }()
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	d.logger.Debug("bpffs mount helper is being called", logger.Ctx{"pid": pid, "bpffsParams": bpffsParams})
+	stdout, err := shared.RunCommandInheritFds(
+		ctx,
+		[]*os.File{pidFd},
+		d.state.OS.ExecPath,
+		"forkmount",
+		"bpffs",
+		"--",
+		fmt.Sprint(pid),
+		fmt.Sprint(pidFdNr),
+		bpffsParams["mountpoint"],
+		bpffsParams["delegate_cmds"],
+		bpffsParams["delegate_maps"],
+		bpffsParams["delegate_progs"],
+		bpffsParams["delegate_attachs"])
+	if err != nil {
+		d.logger.Error("bpffs mount helper has failed", logger.Ctx{"err": err, "stdout": stdout})
+		return err
+	}
+
+	d.logger.Debug("bpffs mount helper has finished without error", logger.Ctx{"stdout": stdout})
+
+	return nil
+}
+
+// onStartHost implements the LXC start-host hook.
+func (d *lxc) onStartHost(args map[string]string) error {
+	if shared.IsFalseOrEmpty(d.expandedConfig["security.delegate_bpf"]) {
+		return nil
```

Comment:
Lets move the contents of his function from line 2580 into its own function, suggest `d.onStartHostBPFDelegate()` or similar, and then reverse this if statement's logic to call that function if `shared.IsTrue(d.expandedConfig["security.delegate_bpf"])`.

That way we nicely structure the general `onStartHost` function with the BPF delegation logic, allowing for easier future expansion for other purposes.

Reply:
Fixed, thanks!